R/logs.R

```diff
@@ -18,6 +26,35 @@ fetch_logs <- function(object, limit = 100, ...) {
 }
 
 #' @export
+fetch_logs.civis_api <- function(object, limit = 100, ...) {
+  f <- find_log_fetcher(name = attr(object, "fname"))
+  args <- attr(object, "args")
+  logs <- do.call(f, c(args, limit = limit))
+  format_scripts_logs(logs)
+}
+
+#' @export
+fetch_logs.civis_error <- function(object, limit = 100, ...) {
+  f <- find_log_fetcher(name = attr(object, "f"))
+  args <- attr(object, "args")
+  logs <- do.call(f, c(args, limit = limit))
+  format_scripts_logs(logs)
+}
+
+find_log_fetcher <- function(name) {
+
+  # This captures slightly more than necessary, but includes all candidates.
+  available <- grep("*_logs$", ls(getNamespace("civis")), value = TRUE)
+
+  # The only exception this rule is models_list_builds_logs, which we don't use.
+  candidate <- paste0(gsub("get", "list", name), "_logs")
+
+  fetcher <- grep(candidate, available, value = TRUE)
+  if (length(fetcher) == 0) stop("No function to fetch logs.")
+  get(fetcher)
```

Comment:
Just a sanity check, but will this work when a function is called with

```r
civis::f(...)
``` 

as well as when a function is called with

```r
library(civis)
f(...)
```

Reply:
```
q <- civis::query_civis("SELECT * FROM datascience.iris")
qrr <- civis::query_civis(q$id)
civis::fetch_logs(qrr)
```
Works!